teps/0140-producing-results-in-matrix.md

```diff
@@ -0,0 +1,702 @@
+```
+pr-build-0 - “image-digest-build-1”
+pr-build-1 - “image-digest-build-2”
+pr-build-2 - “image-digest-build-3”
+```
+
+### Create an object mapping of task run results to be able to filter for a specific combination
+
+Pros: This approach requires the most lift and would enable users to map each result back to a specific combination of params using the syntax `$(tasks.<include-name>.results.<result-name><param.value>-<param.value>)`.
+Cons: This approach might add unnecessary overhead and complexity. There would be limitations with consuming results if the parameter name is unknown. It doesn’t scale with an increasing number of parameters. Another con is that there is a limit with the size of results and non-static values could be coming from results of other tasks where the size is unknown at runtime.
+
+```yaml
+tasks:
+  - name: produce-platforms-and-browsers
+      matrix:
+        params:
+          - name: platform
+            value:
+              - linux
+              - mac
+              - windows
+          - name: browser
+            value:
+              - chrome
+              - safari
+              - firefox
+      taskSpec:
+        params:
+          - name: platform
+            type: string
+          - name: browser
+            type: string
+        results:
+          - name: report-url
+            type: string
+        steps:
+          - name: echo
+            image: alpine
+            script: |
+              echo "https://api.example/get-report/$(params.platform)-$(params.browser)"
+    - name: fetch-linux-reports
+      params:
+        - name: report-url
+          value:
+          - $(tasks.produce-platforms-and-browsers.results.report-url.linux-chrome)
+          - $(tasks.produce-platforms-and-browsers.results.report-url.linx-safari)
+          - $(tasks.produce-platforms-and-browsers.results.report-url.linux-firefox)
+    taskSpec:
+    steps:
+      - name: echo
+        image: alpine
+        script: |
+          ...
+          echo "fetching report: $(results.report-url)"
+```
+
+TaskRuns Created - Result
+```
+pr-produce-platforms-and-browsers-0 - “path/to/report/linux-chrome”
+pr-produce-platforms-and-browsers-1 - “path/to/report/linux-safari”
+pr-produce-platforms-and-browsers-2 - “path/to/report/linux-firefox”
+pr-produce-platforms-and-browsers-3 - “path/to/report/mac-chrome”
+pr-produce-platforms-and-browsers-4 - “path/to/report/mac-safari”
+pr-produce-platforms-and-browsers-5 - “path/to/report/mac-firefox”
+pr-produce-platforms-and-browsers-6 - “path/to/report/windows-chrome”
+pr-produce-platforms-and-browsers-7 - “path/to/report/windows-safari”
+pr-produce-platforms-and-browsers-8 - “path/to/report/windows-firefox”
+```
+
+### Upgrade and Migration Strategy
+
+`Matrix` is still in `alpha` so we can make the proposed changes per the API compatibility policy. We will communicate
+the breaking changes to users in the release notes, and through an email to tekton-dev@ and tekton-users@ if needed.
```

Comment:
what breaking changes does this refer to?

Reply:
Removed this section.